### DISCUSSION TITLE
Caddy Clean URLs don't work

### DISCUSSION THREAD

**Discussion by ndegruchy** [xenocrat/chyrp-lite#258] [General]
Hello,

I'm trying out Chyrp Lite and importing the `caddyfile` into my Caddy config results in an error:

```
Error: adapting config using caddyfile: parsing caddyfile: parsing caddyfile tokens for 'rewrite': /path-to-my-root-directory/caddyfile:25 - Error during parsing: Wrong argument count or unexpected line ending after '/index.php'
```

(yes, the path is correct in my config, which looks like this)

```Caddyfile
mydomain.tld {
  root * /path-to-my-root-directory/

  encode gzip zstd

  @static {
    file
    path *.ico *.css *.js *.gif *.webp *.heic *.avif *.jpg *.jpeg *.png *.svg *.woff *.woff2 *.ogv>
  }
  header @static Cache-Control max-age=5184000

  file_server {
    precompressed br zstd gzip
  }

  import /path-to-my-root-directory/caddyfile
  php_fastcgi unix//var/run/php/php8.2-fpm.sock
} 
```

I'm not sure what is going on here. Without the import, the site works fine, the system just doesn't generate clean URLs. I've tried a couple of times to come at the problem a different way:

```Caddyfile
...
try_files /* {path} {path}/ /index.php
try_files /admin/* {path} {path}/ /admin/index.php
...
```

Nothing seems to work with it, though...

Caddy reports it's version as 2.6.2. I have no `xcaddy` or other compiled in modules. Just the Debian Stable version.

**Comment by xenocrat** [comment]
Hello there,

Any spaces in your Chyrp path? We could try commenting out [line 25](https://github.com/xenocrat/chyrp-lite/blob/master/includes/caddyfile.conf#L25) from the Caddyfile and see if the two other named rewrites are working. Line 25 stops people from being able to view the source of twig files but it’s not essential for clean URLs to function.

**Comment by xenocrat** [comment]
I've just [added a commit](https://github.com/xenocrat/chyrp-lite/commit/6e318eb512160856d15f6c1341a24aba25b64e79) to change the way twig files are rewritten in the Caddyfile. This might fix it. Please test with this updated Caddyfile.conf and let me know.

**Comment by ndegruchy** [comment]
The initial fix seems to work. I didn't get home until late, so I've only been able to do minor testing. I'll make the 'real' fix tomorrow.

Thanks for looking at this, I appreciate it.

**Comment by xenocrat** [comment]
No problem - thank you for reporting it! Hopefully we can get it fixed.

**Comment by ndegruchy** [comment]
Seems to be working well. I would say this is closed/fixed!

Thanks so much!

**Comment by xenocrat** [comment]
Great stuff! Thank you for testing the fix and reporting back. ^_^
